File: framework/data-jpa/src/main/java/top/bettercode/simpleframework/data/jpa/support/QuerySize.java
package top.bettercode.simpleframework.data.jpa.support;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author Peter Wu
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.METHOD, ElementType.ANNOTATION_TYPE})
@Documented
@Inherited
public @interface QuerySize {

  /**
   * @return 查询数量
   */
  int value();
}

File: data-jpa/src/main/kotlin/top/bettercode/summer/data/jpa/support/QuerySize.kt
package top.bettercode.summer.data.jpa.support

import java.lang.annotation.Inherited

/**
 * @author Peter Wu
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.ANNOTATION_CLASS)
@MustBeDocumented
@Inherited
annotation class QuerySize(
        /**
         * @return 查询数量
         */
        val value: Int
) 